api/src/test/cards.test.ts

```diff
@@ -171,4 +123,51 @@ describe('Testing Cards', () => {
       expect(response.body.data.image).toBe(cardData2.image);
     });
   });
+
+  // GET all cards. 
+  describe('[GET] /card', () => {
+    it('response statusCode 200', async () => {
+      const cardsRoute = new CardsRoute();
+      const app = new App([cardsRoute]);
+
+      const result = await request(app.getServer()).get(`${cardsRoute.path}`);
+      expect(result.status).toEqual(200);
+      // Check to see that the there is at least one value.
+      expect(result.body.data.length).toBeGreaterThanOrEqual(1);
+      // Check if the first items id is a length of 24.
+      // Mongo Object ID data types are a 24 character hexadecimal code.
+      expect(result.body.data[0]._id).toHaveLength(24);
+    });
+  });
+
+  // GET card by ID
+  describe('[GET] /card/:id', () => {
+    it('response with card and ID matches', async () => {
+      const cardsRoute = new CardsRoute();
+      const app = new App([cardsRoute]);
+
+      // Card Data to look for.
+      const cardData = {
+        "name": "Card 1",
+        "description": "Description 1",
+        "salePrice": 1,
+        "marketPrice": 1,
+        "rating": [],
+        "image": "linktoImage",
+        "priceHistory": []
+      }
+
+      // Manually create a card (this just adds a card to mongo without using an endpoint)
+      const createdCard: PokemonCard = await PokemonCardModel.create(cardData);
+      // Store the newly generated id.
+      const cardId = createdCard._id;
+      
+      // Pass the new ID to the [GET] by ID endpoint
+      const result = await request(app.getServer()).get(`${cardsRoute.path}/${cardId}`);
+      // Expect good status code and cardId to match the route
+      expect(result.status).toEqual(200);
+      // Note had to use toString because of objectId type.
+      expect(result.body.data._id.toString()).toEqual(cardId.toString());
```

Comment:
I don't think the `result.body.data._id.toString()` needs a .toString(), but it works nonetheless, so it should be fine. Everything looks great